packages/react/src/components/linkbox/Linkbox.tsx

```diff
@@ -0,0 +1,158 @@
+  /**
+   * Aria-label of the linkbox component.
+   */
+  ariaLabel: string;
+  /**
+   * Boolean indicating for external link that takes user to an entirely new web site. Defaults to false.
+   */
+  external?: boolean;
+  /**
+   * Optional heading of the linkbox.
+   */
+  heading?: string;
+  /**
+   * Heading level for linkbox title. Defaults to 2.
+   */
+  headingAriaLevel?: number;
+  /**
+   * The image props passed to the linkbox image. Image variant requires prop withImg set to true.
+   */
+  imgProps?: React.ComponentPropsWithoutRef<'img'>;
+  /**
+   * Boolean for the variant with no background color. Changes paddings also. Defaults to false.
+   */
+  noBackground?: boolean;
+  /**
+   * Aria label for opening link in an external domain. Active when external prop is true.
+   */
+  openInExternalDomainAriaLabel?: string;
+  /**
+   * Boolean indicating whether the link will open in new tab or not. Defaults to false.
+   */
+  openInNewTab?: boolean;
+  /**
+   * Aria label for opening link in a new tab
+   */
+  openInNewTabAriaLabel?: string;
+  /**
+   * Optional text of the linkbox.
+   */
+  text?: string;
+  /**
+   * Boolean indicating whether the Linkbox contains a border or not. Defaults to false.
+   */
+  withBorder?: boolean;
+  /**
+   * Boolean indicating whether the Linkbox contains a hero image or not. Defaults to false.
+   */
+  withImg?: boolean;
+  /**
+   * Size variant for the linkbox when accompanied with an img. Defaults to 'small'
+   */
+  withImgSize?: 'small' | 'medium' | 'large';
+};
+
+export const Linkbox = ({
+  ariaLabel,
+  children,
+  className,
+  external = false,
+  heading,
+  headingAriaLevel = 2,
+  href,
+  imgProps,
+  noBackground = false,
+  openInExternalDomainAriaLabel,
+  openInNewTab = false,
+  openInNewTabAriaLabel,
+  text,
+  withBorder = false,
+  withImg = false,
+  withImgSize = 'small',
+  ...rest
+}: LinkboxProps) => {
+  const linkRef = useRef(null);
+
+  const comboseAriaLabel = () => {
+    const newTabText = openInNewTab ? openInNewTabAriaLabel || 'Avautuu uudessa välilehdessä.' : '';
+    const externalText = external ? openInExternalDomainAriaLabel || 'Siirtyy toiseen sivustoon.' : '';
+    let extendedAriaLabel = '';
+
+    if (ariaLabel && ariaLabel.slice(-1) !== '.') {
+      extendedAriaLabel = `${ariaLabel}.`;
+    }
+
+    return [extendedAriaLabel, newTabText, externalText].filter((txt) => txt).join(' ');
+  };
+
+  return (
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions
+    <div
+      role="region"
+      tabIndex={0} // eslint-disable-line jsx-a11y/no-noninteractive-tabindex
+      onKeyPress={(e) => {
+        if (e.key === 'Enter') {
+          linkRef.current.click();
+        }
+      }}
+      onClick={() => {
+        linkRef.current.click();
+      }}
+      className={classNames(styles.linkbox, withBorder && styles.withBorder, !withBorder && styles.withoutBorder)}
```

Comment:
I had some trouble getting the content appear correctly and same with and without the border. How I managed to fix it was setting a transparent border to the component when there's no border:

`border: solid 2px transparent;`

So that's why there's a separate style for the withoutBorder variant. I checked and this approach was used elsewhere in HDS also. We could set the transparent border style to the basic linkbox style definition, but then we would rely on overwriting when it comes to the withBorder variant. I guess I wanted to avoid that. Another approach that I can think of  would be to create modifiers like:

`classNames(styles.linkboxWithoutBorder, withBorder && styles.linkboxWithBorder)`

Then they both would have to repeat the basic styles for linkbox, and add the difference in border. Though that repetition could be handled nicely with Sass mixins. Would that be better?

Reply:
I have used overwrites pretty often 😁 
The upside there is that those need less code:
```css
.default {
  border: 2px solid transparent;
}

.withBorder {
  border-color: var(--color-coat-of arms);
}
```